src/main/java/com/gregtechceu/gtceu/core/mixins/ItemStackMixin.java

```diff
@@ -0,0 +1,325 @@
+                if (spoilable1 != null && spoilable2 != null) {
+                    cir.setReturnValue(
+                            spoilable1.getTicksUntilSpoiled(stack) == spoilable2.getTicksUntilSpoiled(other));
+                    cir.cancel();
+                }
+            } else {
+                long tick1 = tag1.getLong("creation_tick");
+                long tick2 = tag2.getLong("creation_tick");
+                if (tick1 != tick2) {
+                    long avg;
+                    if (stack.getCount() + other.getCount() > 0)
+                        avg = (tick1 * stack.getCount() + tick2 * other.getCount()) /
+                                (stack.getCount() + other.getCount());
+                    else avg = tick1;
+                    tag1.putLong("creation_tick", avg);
+                    tag2.putLong("creation_tick", avg);
+                }
+            }
+        }
+        cir.setReturnValue(isSameItem && Objects.equals(stack.getTag(), other.getTag()));
+        cir.cancel();
+    }
+
+    @Inject(at = @At(value = "INVOKE",
+                     target = "Lnet/minecraft/world/item/Item;appendHoverText(Lnet/minecraft/world/item/ItemStack;Lnet/minecraft/world/level/Level;Ljava/util/List;Lnet/minecraft/world/item/TooltipFlag;)V"),
+            method = "getTooltipLines",
+            locals = LocalCapture.CAPTURE_FAILSOFT)
+    private void aprilFoolsTooltip(Player player, TooltipFlag isAdvanced, CallbackInfoReturnable<List<Component>> cir,
+                                   List<Component> list) {
+        ISpoilableItem spoilable = ISpoilableItem.getSpoilable((ItemStack) (Object) this);
+        if (!(getItem() instanceof ISpoilableItem) && spoilable instanceof IAddInformation addInformation) {
+            addInformation.appendHoverText((ItemStack) (Object) this, player == null ? null : player.level(), list,
+                    isAdvanced);
+        } else if (gtceu$fakeTooltip) {
+            list.add(Component.translatable(
+                    "gtceu.tooltip.spoil_time_remaining",
+                    Component.literal(FormattingUtil.formatTime(100)).withStyle(ChatFormatting.DARK_AQUA)));
+            list.add(Component.translatable(
+                    "gtceu.tooltip.spoils_into",
+                    Items.DIRT.getDefaultInstance().getDisplayName()));
+        }
+    }
+
+    @Inject(at = @At("HEAD"), method = "isBarVisible", cancellable = true)
+    private void aprilFoolsBarVisible(CallbackInfoReturnable<Boolean> cir) {
+        ISpoilableItem spoilable = ISpoilableItem.getSpoilable((ItemStack) (Object) this);
+        if (!(getItem() instanceof ISpoilableItem) && spoilable instanceof IDurabilityBar durabilityBar) {
+            cir.setReturnValue(durabilityBar.isBarVisible((ItemStack) (Object) this));
+            cir.cancel();
+        } else if (gtceu$fakeTooltip) {
+            cir.setReturnValue(true);
+            cir.cancel();
```

Comment:
doing `cir.setReturnValue` cancels the injected method automatically.
```suggestion
        } else if (gtceu$fakeTooltip) {
            cir.setReturnValue(true);
```

Reply:
Also remove the other instances of this